### PR TITLE
Improve permission checks on several plugins

### PR DESCRIPTION
**Pull request by lumap**
This PR makes it impossible to trigger the edit/reply in read-only channels. Additionally, I moved this plugin to it's own folder and added a README

### CODE REVIEW

**Comment by TheKodeToad** [comment]
QuickReply and SearchReply also have this issue fyi
Also Discord normally lets you press edit on messages in read-only channels, you just can't write anything (iirc)

**Comment by lumap** [comment]
> QuickReply and SearchReply also have this issue fyi
they indeed do, gonna fix that in this pr while i'm at it
> Also Discord normally lets you press edit on messages in read-only channels, you just can't write anything (iirc)
that's probably just a very old low prio bug they never bothered to fix

**Comment by TheKodeToad** [comment]
:nod:

**Comment by lumap** [comment]
QuickMention also had the same issue, fixed it alongside the other two

File: src/plugins/messageClickActions/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# MessageClickActions
+
+Allows you to double click to edit/reply to a message or delete it if you hold the backspace key
+
+![Example](https://i.imgur.com/Ttc905O.gif);
```
Comment (Vendicated) [inline src/plugins/messageClickActions/README.md:5]: avoid imgur, use github

Comment (Vendicated) [reply src/plugins/messageClickActions/README.md:5]: you should also either use a more descriptive alt text or leave it blank

**Comment by Vendicated** [comment]
im pretty sure PermissionBits is a webpack common you can just import, so you should use that instead

also please do not add yourself as plugin author if you're just making such a minor change

**Comment by lumap** [comment]
> im pretty sure PermissionBits is a webpack common you can just import

i did miss that, fixed

File: src/plugins/messageClickActions/index.ts [hunk 2]
```diff
@@ -89,7 +89,7 @@ export default definePlugin({
                         showMentionToggle: channel.guild_id !== null
                     });
                 }
-            } else if (settings.store.enableDeleteOnClick && (isMe || PermissionStore.can(MANAGE_CHANNELS, channel))) {
+            } else if (settings.store.enableDeleteOnClick && (isMe || PermissionStore.can(PermissionsBits.MANAGE_CHANNELS, channel))) {
```
Comment (Vendicated) [inline src/plugins/messageClickActions/index.ts:92]: uhh, this should be MANAGE_MESSAGES not CHANNELS

idk what i was cooking there

**Comment by Vendicated** [comment]
thanks!

**Pull request merged**